pkg/collector/corechecks/gpu/nvmlmetrics/collector.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+// Package nvmlmetrics holds the logic to collect metrics from the NVIDIA Management Library (NVML).
+// The main entry point is the Collector struct, which is responsible for collecting metrics from the NVML library.
+// There are multiple subsystems that can be used to collect metrics from NVML: GPM, FieldInfo functions, and general
+// API functions. The collection of metrics is done by the subsystems, implementing the subsystemCollector interface.
+package nvmlmetrics
+
+import (
+	"fmt"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+	"github.com/hashicorp/go-multierror"
+)
+
+// Collector is the main struct responsible for collecting metrics from the NVML library.
+type Collector struct {
```

Comment:
should we use here the prometheus collector interface?
(or our [wrapper](https://github.com/DataDog/datadog-agent/blob/main/comp/core/telemetry/collector.go) for it)

Reply:
I don't think it's a useful abstraction in this case, as they work on channels and here we're directly collecting the metrics.